packages/ui/src/components/item-layout/item-layout.native.tsx

```diff
@@ -0,0 +1,103 @@
+import { ReactNode, isValidElement } from 'react';
+import { StyleSheet } from 'react-native';
+
+import { Flag } from '../../components/flag/flag.native';
+import { Text } from '../../components/text/text.native';
+import { ChevronUpIcon } from '../../icons/chevron-up-icon.native';
+import { Box } from '../box/box.native';
+
+export interface ItemLayoutProps {
+  captionLeft: ReactNode;
+  captionRight?: ReactNode;
+  flagImg: ReactNode;
+  isDisabled?: boolean;
+  isSelected?: boolean;
+  showChevron?: boolean;
+  titleLeft: ReactNode;
+  titleRight: ReactNode;
+}
+
+export function ItemLayout({
+  captionLeft,
+  captionRight,
+  flagImg,
+  showChevron,
+  titleLeft,
+  titleRight,
+}: ItemLayoutProps) {
+  return (
+    <Flag img={flagImg} spacing="3">
+      <Box style={styles.container}>
+        <Box
+          style={[
+            styles.stack,
+            {
+              alignItems: 'flex-start', // this start on web
+              justifyContent: 'space-between',
+              flexGrow: 2,
+              overflow: 'hidden',
+            },
+          ]}
+        >
+          <Box style={styles.hstack}>
+            {/* how to handle text overflow? or do i need to */}
+            isValidElement(titleLeft) ? titleLeft : <Text>{titleLeft}</Text>
+          </Box>
+          isValidElement(captionLeft) ? captionLeft :
+          <Text variant="caption01" style={[styles.pressableCaptionStyles]}>
+            {captionLeft}
+          </Text>
+        </Box>
+        <Box style={[styles.hstack, { gap: 3 }]}>
+          <Box
+            style={[
+              styles.stack,
+              {
+                alignItems: 'flex-end', // this end on web
+                gap: 2,
+                height: 42,
+              },
+            ]}
+          >
+            {isValidElement(titleRight) ? titleRight : <Text variant="label02">{titleRight}</Text>}
+            {isValidElement(captionRight) ? (
+              captionRight
+            ) : (
+              <Text variant="caption01" style={[styles.pressableCaptionStyles]}>
+                {captionRight}
+              </Text>
+            )}
+          </Box>
+          {showChevron && (
+            <ChevronUpIcon
+              color="ink.action-primary-default"
+              transform="rotate(90deg)"
+              variant="small"
+            />
+          )}
+        </Box>
+      </Box>
+    </Flag>
+  );
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    alignItems: 'center',
+    justifyContent: 'space-between',
+  },
+  stack: {
+    flex: 1,
+    flexDirection: 'column',
+    alignItems: 'center',
+  },
+  hstack: {
+    flex: 1,
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  pressableCaptionStyles: {
```

Comment:
Adding this on web as we have:
```
export const pressableCaptionStyles = css({
  _groupDisabled: { color: 'ink.text-non-interactive' },
  color: 'ink.text-subdued',
});

export const pressableChevronStyles = css({
  _groupDisabled: { color: 'ink.text-non-interactive' },
  color: 'ink.action-primary-default',
});
```

I think maybe we can just add `color` only instead as `_groupDisabled` doesn't seem to apply here

Reply:
_groupDisabled is from panda: https://panda-css.com/docs/concepts/conditional-styles#group-selectors

Reply:
Thanks @fbwoolf . I'll exclude it for now and we can assess how it looks without, is that OK?

Reply:
O yeah, totally fine, was just letting @edgarkhanzadian know where it came from. It was used to style the children based on the parent being disabled. Not sure that is even relevant in mobile?